src/util/check.h

```diff
@@ -80,4 +86,23 @@ T&& inline_assertion_check(T&& val, [[maybe_unused]] const char* file, [[maybe_u
  */
 #define Assume(val) inline_assertion_check<false>(val, __FILE__, __LINE__, __func__, #val)
 
+/**
+ * NONFATAL_UNREACHABLE() is a macro that is used to mark unreachable code. It throws a NonFatalCheckError.
+ * This is used to mark code that is not yet implemented or is not yet reachable.
+ */
+#define NONFATAL_UNREACHABLE()                                        \
+    throw NonFatalCheckError(                                         \
+        format_internal_error("Unreachable code reached (non-fatal)", \
+                              __FILE__, __LINE__, __func__, PACKAGE_BUGREPORT))
+
+/**
+ * UNREACHABLE is a macro that is used to mark unreachable code. It aborts the program.
+ * This is used to mark code that is not yet implemented or is not yet reachable.
+ */
+#define UNREACHABLE()                                                                                                                                   \
```

Comment:
This is unused and on a second thought I wonder if there is any value in it. The current code can already use `assert(false)` just fine. Introducing another way to write `assert(false)` will just lead to bike-shedding without any benefit.

Reply:
in favor to have a macro like `UNREACHABLE` is that we can the possibility to add more failures to the error message, that it is always good.

Maybe somethings `assert(false && "err message")` can be a good motivation to have this macros, otherwise I agree with @MarcoFalke

Reply:
I think that this is unreachable code, so by definition it is not something that a user will (or should) ever see. So there is not much benefit in spending time to overengineer error messages and formatting of them.

Reply:
> I think that this is unreachable code, so by definition it is not something that a user will (or should) ever see.

It could be useful for developers modifying code and stumbling across an unreachable block as a result.
Though I agree that it's not an essential feature considering `assert(false)` works alright.

Reply:
Developers would already know what `assert(false)` means and what it does. Printing `PACKAGE_BUGREPORT` for them also doesn't seem to add much value.

Reply:
Makes sense, I removed it